### PR TITLE
Docs/fix doxy

### PR DESCRIPTION
**Pull request by jsmith1024**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,5 +33,5 @@ It is a general purpose statically typed, easy to learn and use language.
 #### Manuals:
 - [User Manual](https://jsmith1024.github.io/bill/userman/html/index.html)
 - [Reference Manual](https://jsmith1024.github.io/bill/refman/html/index.html)
-- [Bill_Primer](https://jsmith1024.github.io/bill/langman/html/index.html)
-- [Standard_Library_Refence](https://jsmith1024.github.io/bill/stlman/html/index.html)
+- [Bill Primer](https://jsmith1024.github.io/bill/langman/html/index.html)
```
Comment (nekenyu) [inline README.md]: Are these links still correct?

File: docs/index.html [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
 <html xmlns="http://www.w3.org/1999/xhtml">
     <head>
-        <meta http-equiv="Refresh" content="0; url='html/index.html'" />
+        <meta http-equiv="Refresh" content="0; url='../home/html/index.html'" />
```
Comment (nekenyu) [inline docs/index.html]: This seems wrong.  You want to redirect from docs/index.html to home/html/index.html?  I would think after the last Pull Request, **this** file was generated.  Further, I don't see a home anywhere in your branch.

File: doxy [hunk 3]
```diff
@@ -3,122 +3,52 @@
...
-
-doxyfile="dox/DoxyUserman"
-if [ -f ${doxyfile} ]
-	then
-		doxygen ${doxyfile}
-		retVal=$?
-		if [ $retVal -ne 0 ]
-		then
-            echo "Error: exit code ${retVal} on ${doxyfile}."
+            echo "Error: exit code ${$retVal} on ${doxyfile}."
```
Comment (nekenyu) [inline doxy:65]: Whitespace (indentation)

File: doxy [hunk 4]
```diff
@@ -3,122 +3,52 @@
...
+books=(    "Bill_Primer" "reference_manual"      "user_manual"      "standard_library_reference")
+names=(    "Bill_Primer" "Bill_Reference_Manual" "Bill_User_Manual" "Bill_Standard_Library_Reference")
 
+# html forward and config
+cp dox/index.html  docs/index.html
+cp dox/_config.yml docs/_config.yml
 
+# html docs
 doxyfile="dox/DoxyHome"
 if [ -f ${doxyfile} ]
```
Comment (nekenyu) [inline doxy:59]: Consider an error if the dox file is missing.  But, this 11 lines could be the following in make:
```
.PHONY: docs-html
docs-html:
<tab>: doxygen dox/DoxyHome
```

File: notes/index.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
```
Comment (nekenyu) [inline notes/index.html]: Why is this file needed?

File: doxy [hunk 6]
```diff
@@ -3,122 +3,52 @@
...
-# 		doxygen ${doxyfile}
-# fi
-
-# doxyfile="dox/DoxyMan"
-# if [ -f ${doxyfile} ]
-# 	then
-# 		doxygen ${doxyfile}
-# fi
-
+doxyfiles=("DoxyLangman" "DoxyRefman"            "DoxyUserman"      "DoxyStlman")
```
Comment (nekenyu) [inline doxy]: I know what you're doing here, but it kills me to see multiple arrays.  Consider....  Making one array of "location;doxyfile;name" or something, then splitting the string apart in the loop.....

File: doxy [hunk 7]
```diff
@@ -3,122 +3,52 @@
...
+    if [ -f ${doxyfile} ]
+        then
+            doxygen ${doxyfile}
+            retVal=$?
+            if [ $retVal -ne 0 ]
+            then
+                echo "Error: exit code ${$retVal} on doxygen ${doxyfile}."
+                exit $retVal
+            fi
+            cd /tmp/bill/${locations[i]}/latex
```
Comment (nekenyu) [inline doxy]:
```
pushd /tmp/bill/.....
....
popd
```

File: doxy [hunk 8]
```diff
@@ -3,122 +3,52 @@
...
+            make
+            retVal=$?
+            if [ $retVal -ne 0 ]
+            then
+                echo "Error: exit code ${retVal} on Make of ${doxyfile}."
+                exit $retVal
```
Comment (nekenyu) [inline doxy]: You need to fix directory location before you exit.

File: doxy [hunk 9]
```diff
@@ -3,122 +3,52 @@
...
+            fi
+            cd $here
+            mv /tmp/bill/${locations[i]}/latex/refman.pdf docs/${names[i]}.pdf
+    else
+        echo -e "\n========\nInfo: ${doxyfile} does not exist.\n========\n"
+    fi
```
Comment (nekenyu) [inline doxy]: Consider terminate with error

**Review by nekenyu — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jsmith1024) [reply README.md]: yes

File: doxy (hunk 9 above)
Comment (jsmith1024) [reply doxy]: Info was chosen, since it's current absence is intended.

File: doxy (hunk 6 above)
Comment (jsmith1024) [reply doxy]: Given, that's an area of shell scrip I've never messed with, could we relegate it to an issue for now?

Comment (nekenyu) [reply doxy]: Okay, I have a better idea.  Why don't you make a shell script function that takes the relevant arguments, then instead of making arrays, you call the function sequentially with the right arguments?  That is probably easier.

File: docs/index.html (hunk 2 above)
Comment (jsmith1024) [reply docs/index.html]: Fixed.

File: doxy (hunk 3 above)
Comment (jsmith1024) [reply doxy:65]: Please elaborate.

File: doxy (hunk 8 above)
Comment (jsmith1024) [reply doxy]: fixed.

File: notes/index.html (hunk 5 above)
Comment (jsmith1024) [reply notes/index.html]: What on Earth, I deleted it already.???

File: README.md (hunk 1 above)
Comment (nekenyu) [reply README.md]: I don't see how they can work when the top level directory does not have directories "userman" "refman" and "langman" on the branch.

Comment (jsmith1024) [reply README.md]: Oh, I overlooked that...

File: doxy [hunk 10]
```diff
@@ -1,18 +1,60 @@
...
+    # build if DoxyFile exists
+    doxyfile="dox/books/${book}/${config}"
+    if [ -f ${doxyfile} ]
+    then
+        doxygen ${doxyfile}
+        retVal=$?
+        if [ $retVal -ne 0 ]
+        then
+            echo "Error: exit code ${retVal} on doxygen ${doxyfile}."
+#                 cd $here
```
Comment (nekenyu) [inline doxy:26]: Please remove this comment.

File: doxy [hunk 11]
```diff
@@ -1,18 +1,60 @@
...
+        echo -e "\n========\nInfo: ${doxyfile} does not exist.\n========\n"
+        # future activation
+#         echo -e "\n========\nError: ${doxyfile} does not exist.\n========\n"
+#         popd
+#         exit 1
+    fi
+}
 
-# html forward and config
+# copy html forward and config
```
Comment (nekenyu) [inline doxy:53]: Do these get overwritten by doxygen?  Or can you just git mv them and remove the copies in the script?

File: README.md [hunk 12]
```diff
@@ -31,7 +31,7 @@ It is a general purpose statically typed, easy to learn and use language.
 - strong static typing
 
 #### Manuals:
-- [User Manual](https://jsmith1024.github.io/bill/userman/html/index.html)
-- [Reference Manual](https://jsmith1024.github.io/bill/refman/html/index.html)
-- [Bill_Primer](https://jsmith1024.github.io/bill/langman/html/index.html)
-- [Standard_Library_Refence](https://jsmith1024.github.io/bill/stlman/html/index.html)
+- [User Manual](https://jsmith1024.github.io/bill/User.html)
```
Comment (nekenyu) [inline README.md]: I really think these are .../bill/html/User.html and similar.

File: README.md (hunk 1 above)
Comment (nekenyu) [reply README.md]: Resolving conversation and creating a new one because the code is detached from the conversation.

File: doxy (hunk 4 above)
Comment (nekenyu) [reply doxy:59]: I still don't see an error if the main doxyfile is missing.

Comment (jsmith1024) [reply doxy:59]: It does need one. I made one. I don't want errors going off right now, when I know they aren't created yet. So I commented it out, with a note to activate later.

File: doxy (hunk 11 above)
Comment (jsmith1024) [reply doxy:53]: I do it this way to keep all configuration in dox.

Comment (nekenyu) [reply doxy:53]: I don't like that answer on principle, bu given that we are committing the docs directory for the website....  I guess that makes clear that everything in docs/ is temporary for the website and all source (not so much configuration) is in dox.

File: doxy (hunk 4 above)
Comment (nekenyu) [reply doxy:59]: This is with respect to the html.  If that file is missing (which for html should not be missing), you have a problem and you won't detect it.

**Review by nekenyu — APPROVED** [review]
